baseplate/thrift_pool.py

```diff
@@ -37,13 +37,22 @@
 logger = logging.getLogger(__name__)
 
 
-def _make_protocol(endpoint):
+def _make_protocol(endpoint, transport_wrapper=None):
+    """
+    The transport_wrapper parameter passed in here should be an instance of a
+    factory that allows us to wrap the TSocket implementation in a
+    BufferedTransport or FramedTransport if necessary.
+    """
     if endpoint.family == socket.AF_INET:
         trans = TSocket.TSocket(*endpoint.address)
     elif endpoint.family == socket.AF_UNIX:
         trans = TSocket.TSocket(unix_socket=endpoint.address)
     else:
         raise Exception("unsupported endpoint family %r" % endpoint.family)
+
+    if transport_wrapper is not None:
```

Comment:
i must have missed something getting lost in THeaderProtocol. cool! :)

i think this would probably be a cleaner API though if that's cool